### PR TITLE
Add helper scripts for preparing GCE environment.

### PR DESCRIPTION
**Pull request by yungyuc**
Port the helper scripts in https://github.com/solvcon/solvcon-gce to Python.  And bring them into this main repository to simplify GCE management.

I tested the script for several days and it works fine.  But it is still an early version and considered a prototype.  I place it in the `contrib` directory and hope it's more convenient than the original shell scripts.

### CODE REVIEW

File: contrib/gce/sgc [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+        tokens = " ".join(tokens).split()
+        subprocess.check_call(tokens)
+
+
+class _SubCommandFormatter(argparse.ArgumentDefaultsHelpFormatter,
+                           argparse.RawDescriptionHelpFormatter):
+    pass
+
+
+class _CreateInstance(object):
```
Comment (yungyuc) [inline contrib/gce/sgc:51]: This command creates an instance using either (i) upstream Ubuntu 14.04 LTS image or (ii) a custom image created by this command.

File: contrib/gce/sgc [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+        if args.homedisk:
+            tokens.append("--disk=device-name=sdb,mode=rw,name=%s"
+                          % args.homedisk)
+
+        _call_command(tokens, args.verbose, args.dry_run)
+
+        os.unlink(startup_fn)
+
+
+class _MakeImage(object):
```
Comment (yungyuc) [inline contrib/gce/sgc:216]: This command creates an image from the boot disk created from _CreateInstance command.

**Review by tai271828 — COMMENTED** [review]
Hi, I am in the traveling status and don't have stable network and a chunk of time to have a completed review and trial. After a quick look and the corresponding reflection as the following:

- Quote from [Google Shell Style Guide](https://google.github.io/styleguide/shell.xml) **If you are writing a script that is more than 100 lines long, you should probably be writing it in Python instead. Bear in mind that scripts grow. Rewrite your script in another language early to avoid a time-consuming rewrite at a later date.** Thus I am always +1 to this kind of porting.

- This PR does not remove any existing scripts e.g. https://github.com/solvcon/solvcon-gce/blob/master/bin/gce-create-instance. I suppose this is for compatibility of the other scripts? Some scripts should be ported as well once we are ok to use this prototype python script to replace the existing script. Please let me know if I misunderstand.

- Because this PR did not change the other existing scripts and components, it is fine to merge it if you want to take a further step soon based on this PR. Or I will review and test it in more details later when I finish my trip.

**Comment by yungyuc** [comment]
It's intentional to leave solvcon-gce scripts intact because the new Python script doesn't handle cluster building.  I simply ported the features that I found myself keep using.

I'll merge this PR.  When we ported all the features to this script we may deprecated the old solvcon-gce.

**Pull request merged**